### PR TITLE
set pod memory and cpu based on group definition

### PR DESCRIPTION
**Pull request by nonsense**
1. Fixing a bug where we don't set the pod CPU and memory based on definition, but we are still using the defaults.

2. Reducing the CPU / memory limits for the `mkdir-outputs` container. It exits right at the start of execution, but no point requesting the defaults (which are 100m and 256Mi).

### CODE REVIEW

File: pkg/runner/cluster_k8s.go [hunk 1]
```diff
@@ -733,6 +751,12 @@ func (c *ClusterK8sRunner) createTestplanPod(ctx context.Context, podName string
 							MountPropagation: &mountPropagationMode,
 						},
 					},
+					Resources: v1.ResourceRequirements{
+						Limits: v1.ResourceList{
+							v1.ResourceMemory: resource.MustParse("10Mi"),
```
Comment (coryschwartz) [inline pkg/runner/cluster_k8s.go:756]: I can't think of any reason to have more than 10Mi for a container that just creates a directory.Looks okay to me.

File: pkg/runner/cluster_k8s.go [hunk 2]
```diff
@@ -219,6 +219,24 @@ func (c *ClusterK8sRunner) Run(ctx context.Context, input *api.RunInput, ow *rpc
 				}},
 		})
 
+		podCPU := defaultCPU
```
Comment (coryschwartz) [inline pkg/runner/cluster_k8s.go:222]: just a style thing, but we have another way for deriving default configs in the docker runner. Not sure why we use mergo.Merge there and just if-statements in the k8s-runner.

File: pkg/runner/cluster_k8s.go (hunk 1 above)
Comment (nonsense) [reply pkg/runner/cluster_k8s.go:756]: Probably less is fine as well, but then again this is freed as soon as the container completes, so it is just for `scheduling` calculations.

File: pkg/runner/cluster_k8s.go (hunk 2 above)
Comment (nonsense) [reply pkg/runner/cluster_k8s.go:222]: I will check how we do that in the local runner and see if we can align in a future PR.

**Pull request merged**